src/PHPStan/Rules/NoMixedInArrayShapeRule.php

```
@@ -0,0 +1,81 @@
+
+        // Check for mixed in array type annotations
+        // Patterns: array<string, mixed>, array<mixed>, list<mixed>, etc.
+        if (preg_match('/array\s*<[^>]*\bmixed\b[^>]*>/', $text)) {
+            $errors[] = RuleErrorBuilder::message(
+                'Array type contains "mixed". Use clear array shapes (e.g. array{key: string}) and/or DTOs.',
+            )->identifier('noMixed.arrayShape')->build();
+        }
+
+        // Check for mixed in @param annotations
+        if (preg_match_all('/@param\s+[^\s]*\bmixed\b/', $text, $matches)) {
+            foreach ($matches[0] as $match) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$match'. (undefined)

(UnusedLocalVariable)